File: Kotlin/그리디 알고리즘/15904.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import kotlin.system.exitProcess

fun main() = with(BufferedReader(InputStreamReader(System.`in`))){
    val s = readLine()
    val UCPC = "UCPC"
    var idx = 0
    for (i in UCPC){
        val temp = s.indexOf(i, idx)
        if (temp == -1){
            println("I hate UCPC")
            exitProcess(0)
        }
        idx = temp
    }
    println("I love UCPC")

}

File: BOJ/Python/그리디 알고리즘/15904.py
S = input()
UCPC = "UCPC"
idx = 0
for i in UCPC:
    temp = S.find(i, idx)
    if temp == -1:
        print("I hate UCPC")
        break
    idx = temp
else:
    print("I love UCPC")
